internal/provider/provider.go

```diff
@@ -224,11 +224,23 @@ func New(version string) func() provider.Provider {
 	}
 }
 
-func (p *NuoDbaasProvider) ConfigValidators(context.Context) []provider.ConfigValidator {
-	return []provider.ConfigValidator{
-		providervalidator.RequiredTogether(
-			path.MatchRoot("user"),
-			path.MatchRoot("password"),
-		),
+func (p *NuoDbaasProvider) ValidateConfig(ctx context.Context, req provider.ValidateConfigRequest, resp *provider.ValidateConfigResponse) {
+	var config NuoDbaasProviderModel
+	if !framework.ReadResource(ctx, &resp.Diagnostics, req.Config.Get, &config) {
+		return
+	}
+
+	hasUser := config.GetUser() != ""
+	hasPassword := config.GetPassword() != ""
+
+	if (hasUser && !hasPassword) || (hasPassword && !hasUser) { // user xnor password
+		resp.Diagnostics.AddError("Partial credentials", "To use authenticantion, both user name and password should be provided.")
+	}
+
+	if hasUser {
+		userParts := strings.Split(config.GetUser(), "/")
+		if len(userParts) != 2 || len(userParts[0]) < 1 || len(userParts[1]) < 1 {
+			resp.Diagnostics.AddAttributeError(path.Root("user"), "Malformed user name", "User name should be in the format \"<organization>/<user>\".")
```

Comment:
In the errors above, I just say "Invalid provider configuration", because the `user` attribute will already appear in the message.

Reply:
Why is it better? It is longer and less descriptive.